src/linker/step.py

```diff
@@ -0,0 +1,48 @@
+from pathlib import Path
+from typing import Callable, List
+
+from linker.implementation import Implementation
+
+
+class Step:
+    IMPLEMENTATIONS = {
```

Comment:
Is this the correct place for this?

Reply:
I think it's okay. As I've mentioned before, in the long run I want to have the reference go the other way: an implementation knows what step it implements, as opposed to a step knowing about _all_ implementations of itself.

Reply:
If that's the way we want to go, we should design it that way now.

Reply:
Ok, if so I'll motivate it a bit more: my thinking is that it should be possible for some third party to publish an implementation (e.g. on GitHub or similar) and have it immediately be usable, without having to add it to an "official registry" in the tool itself.